From 06d2e46f08ae64ad0d695b9fb6b38dc1860f84d1 Mon Sep 17 00:00:00 2001
From: Lawrence Rust <lvr@softsystem.co.uk>
Date: Sun, 7 Dec 2014 20:25:09 +0000
Subject: [PATCH 278/333] MediaMon: Avoid an abort in libudf when a data CD is
 inserted

Some versions of libudf can cause a glibc abort in udf_fopen
when a data CD is inserted.
Mythbuild.sh includes a patch to libcdio-0.82/lib/udf/udf_fs.c
to workaround the bugs.
However, when run with standard shared libs there needs to
be a way to avoid the abort.

This patch conditionally excludes the problematic libudf calls
if the shell variable MYTH_NOUDF is defined.

Signed-off-by: Lawrence Rust <lvr@softsystem.co.uk>
---
 mythtv/libs/libmythbase/mythcdrom-linux.cpp |   10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/mythtv/libs/libmythbase/mythcdrom-linux.cpp b/mythtv/libs/libmythbase/mythcdrom-linux.cpp
index 17ae3c3..e47a560 100644
--- a/mythtv/libs/libmythbase/mythcdrom-linux.cpp
+++ b/mythtv/libs/libmythbase/mythcdrom-linux.cpp
@@ -11,6 +11,7 @@
 #include <limits.h>
 #include <linux/iso_fs.h>
 #include <unistd.h>
+#include <stdlib.h>         // getenv
 
 #include <QDateTime>
 
@@ -600,6 +601,14 @@ MythMediaStatus MythCDROMLinux::checkMedia()
                 LOG(VB_MEDIA, LOG_INFO,
                     QString("Volume ID: %1").arg(m_VolumeID));
 #ifdef USING_LIBUDF
+                // Some versions of libudf can cause a glibc abort in udf_fopen
+                // when a data CD is inserted.
+                // Mythbuild.sh includes a patch to libcdio-0.82/lib/udf/udf_fs.c
+                // to workaround the bugs.
+                // However, when run with standard shared libs there needs to
+                // be a way to avoid the abort.
+                if (!std::getenv("MYTH_NOUDF"))
+                {
                 // Check for a DVD/BD disk by reading the UDF root dir.
                 // This allows DVD's to play immediately upon insertion without
                 // calling mount, which either needs pmount or changes to fstab.
@@ -625,6 +634,7 @@ MythMediaStatus MythCDROMLinux::checkMedia()
                         return setStatus(MEDIASTAT_USEABLE, OpenedHere);
                     }
                 }
+                }
 #endif
                 // the base class's onDeviceMounted will do fine
                 // grained detection of the type of data on this disc
-- 
1.7.9.5

